Keas.Core/Services/IEmailService.cs

```diff
@@ -70,200 +51,145 @@ public async Task SendExpiringMessage(int personId, ExpiringItemsEmailModel mode
                 var users = await GetUsersInRoles(roles, person.TeamId);
-                foreach (var user in users)
-                {
-#if DEBUG
-                    Console.WriteLine(user.Email);
-#else
-                    message.CC.Add(user.Email);
-#endif
-                }
+                ccUsers.AddRange(users);
             }
 
             if (expiringItems.KeySerials.Any())
             {
                 var roles = await _dbContext.Roles
                     .Where(r => r.Name == Role.Codes.DepartmentalAdmin || r.Name == Role.Codes.KeyMaster).ToListAsync();
                 var users = await GetUsersInRoles(roles, person.TeamId);
-                foreach (var user in users)
-                {
-#if DEBUG
-                    Console.WriteLine(user.Email);
-#else
-                    message.CC.Add(user.Email);
-#endif
-                }
+                ccUsers.AddRange(users);
             }
 
             if (expiringItems.Equipment.Any())
             {
                 var roles = await _dbContext.Roles
                     .Where(r => r.Name == Role.Codes.DepartmentalAdmin || r.Name == Role.Codes.EquipmentMaster).ToListAsync();
                 var users = await GetUsersInRoles(roles, person.TeamId);
-                foreach (var user in users)
-                {
-#if DEBUG
-                    Console.WriteLine(user.Email);
-#else
-                    message.CC.Add(user.Email);
-#endif
-                }
+                ccUsers.AddRange(users);
             }
 
             if (expiringItems.Workstations.Any())
             {
                 var roles = await _dbContext.Roles
                     .Where(r => r.Name == Role.Codes.DepartmentalAdmin || r.Name == Role.Codes.SpaceMaster).ToListAsync();
                 var users = await GetUsersInRoles(roles, person.TeamId);
-                foreach (var user in users)
-                {
-#if DEBUG
-                    Console.WriteLine(user.Email);
-#else
-                    message.CC.Add(user.Email);
-#endif
-                }
+                ccUsers.AddRange(users);
+                
             }
 
-            message.Subject = "PEAKS Notification";
-            message.IsBodyHtml = false;
-            try
-            {
-                message.Body = await engine.CompileRenderAsync("/EmailTemplates/_Expiring.cshtml", expiringItems);
-            }
-            catch (Exception e)
-            {
-                Console.WriteLine(e.Message);
-                throw;
-            }
+            // transform to cc recipient
+            var ccEmails = ccUsers
+                .Distinct()
+                .Select(u => new Recipient() {Address = new Address(u.Email, u.Name, "cc")})
+                .ToList();
+
+#if !DEBUG
+            // add emails to
+            ccEmails.ForEach(transmission.Recipients.Add);
+#endif
+
+            // build view
+            var engine = GetRazorEngine();

```

Comment:
@jpknoll Did you test this? We had to jump through some hoops using RazorLightEngineBuilder.